scalding-parquet/src/main/scala/com/twitter/scalding/parquet/tuple/macros/impl/WriteSupportProvider.scala

```diff
@@ -0,0 +1,139 @@
+package com.twitter.scalding.parquet.tuple.macros.impl
+
+import com.twitter.bijection.macros.impl.IsCaseClassImpl
+import parquet.io.api.RecordConsumer
+import parquet.schema.MessageType
+
+import scala.reflect.macros.Context
+
+object WriteSupportProvider {
+
+  def toWriteSupportImpl[T](ctx: Context)(implicit T: ctx.WeakTypeTag[T]): ctx.Expr[(T, RecordConsumer, MessageType) => Unit] = {
+    import ctx.universe._
+
+    if (!IsCaseClassImpl.isCaseClassType(ctx)(T.tpe))
+      ctx.abort(ctx.enclosingPosition,
+        s"""We cannot enforce ${T.tpe} is a case class,
+            either it is not a case class or this macro call is possibly enclosed in a class.
+            This will mean the macro is operating on a non-resolved type.""")
+
+    def matchField(idx: Int, fieldType: Type, fValue: Tree, groupName: TermName): (Int, Tree) = {
+      def writePrimitiveField(wTree: Tree) =
+        (idx + 1, q"""rc.startField($groupName.getFieldName($idx), $idx)
+                      $wTree
+                      rc.endField($groupName.getFieldName($idx), $idx)""")
+
+      def writeGroupField(subTree: Tree) =
+        q"""rc.startField($groupName.getFieldName($idx), $idx)
+            rc.startGroup()
+            $subTree
+            rc.endGroup()
+            rc.endField($groupName.getFieldName($idx), $idx)
+         """
+      def writeCollectionField(elementGroupName: TermName, subTree: Tree) =
+        writeGroupField(q"""if(!$fValue.isEmpty) {
+                              val $elementGroupName = $groupName.getType($idx).asGroupType.getType(0).asGroupType
+                              $subTree
+                            }
+                         """)
+
+      fieldType match {
+        case tpe if tpe =:= typeOf[String] =>
+          writePrimitiveField(q"rc.addBinary(Binary.fromString($fValue))")
```

Comment:
Hi @johnynek , this is fixed in this commit: https://github.com/JiJiTang/scalding/commit/60e7390ea018f291728aab65414e125108541e79 merged with PR #1303